docs/developer.qmd

```diff
@@ -2,40 +2,121 @@
+Within the REPL type `]` to enter the Pkg REPL.
+For more information on how to use `Pkg`, see the [Getting Started](https://pkgdocs.julialang.org/v1/getting-started/) page in its documentation.
+There you can add `Revise` and `IJulia` to your global environment.
+
+```julia
+pkg> add Revise IJulia
+```
+
+## Setup Revise.jl
+
+Revise.jl is a library that allows you to modify code and use the changes without restarting Julia. 
+You can let it start automatically by following these [instructions](https://timholy.github.io/Revise.jl/stable/config/#Using-Revise-by-default-1).
+
+## Install Quarto
+
+Quarto is the publishing system that builds our documentation.
+The installation instructions can be found on their [web page](https://quarto.org/docs/get-started/).
+
+## Clone Ribasim
+
+In order to have the Ribasim repository locally available, run the following command at a directory of your choice:
+
+```bash
+git clone https://github.com/Deltares/Ribasim.jl.git
+```
+
+
+## Install Visual Studio Code (optional)
+
+There is a section on editors and IDEs for Julia on <https://julialang.org/>, scroll down to see it. 
+We use and recommend Microsoft's free editor [Visual Studio Code](https://code.visualstudio.com/). 
+When combined with the [Julia extension](https://www.julia-vscode.org/) it provides a powerful and interactive development experience.
+
+
+
+# Developing on Ribasim
+
+## Running tests
+
+You will want to run the testsuite on a regular basis to check if your changes had unexpected side effects.
+It is also a good way to find out if your development environment is set up correctly.
+While not technically required, it is advised to import Ribasim first to get installation issues early on.
+
+```julia
+julia> using Ribasim
+```
+
+Then open the Pkg REPL by typing `]` and execute:
+
+```julia
+pkg> test
+```
+
+If you are in a different environment that depends on Ribasim (like the `run` environment mentioned [later](./developer.html#run-ribasim-simulations)), you can test Ribasim with the following command:
+
+```julia
+pkg> test Ribasim
+```
+
+:::{.callout-tip}
+The Julia VS Code extension allows you to [open a REPL](https://www.julia-vscode.org/docs/stable/userguide/runningcode/#The-Julia-REPL) with the Ribasim environment already activated.
+That way you don't have to type `using Ribasim` every time you open the REPL.
+:::
+
+## Render documentation
+
+In order to preview documentation you can run the following command at the root of this project.
+Afterwards, a browser tab will open with the rendered documentation.
+
+```bash
+quarto preview docs
+```
+
+The documentation also includes [Jupyter notebooks](https://jupyter.org/).
+To edit and execute them run the following commands in your Julia REPL.
+The first time, it will offer to download Conda and Jupyter for you.
+Say yes, and a browser tab will open with Jupyter loaded.
+
+```julia
+julia> using IJulia
+julia> jupyterlab(".")
```

Comment:
```suggestion
julia> jupyterlab(dir=".")
```

It's a keyword argument, and unlike Python providing it as a positional one doesn't work.

Reply:
Oh good catch, I should have tried the command first.